### PR TITLE
Fix crashes in OpenGLVersionChecker 

### PR DESCRIPTION
**Pull request by samcake**
Add more safe checks on the values returned from the opengl context creation to avoid crashing on non capable hardwares

THanks to @davidkelly for pointing out the issue

### CODE REVIEW

**Comment by samcake** [comment]
Test Plan:
Try running this PR build on a PC without a gpu or with very low openGL support (under 4.1) and make sure that it never crashes. during that check.
A dialog box explaining that the current hardware is under the minimum specs should be displayed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 86a569651c501da4f4f3e7d70e809b8023b67135 available at: https://gist.github.com/cfaaec64d26b4985982aee1bc347e78c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-86a5696.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-86a5696.dmg

File: libraries/gl/src/gl/OpenGLVersionChecker.cpp [hunk 1]
```diff
@@ -75,15 +75,26 @@ QJsonObject OpenGLVersionChecker::checkVersion(bool& valid, bool& override) {
...
-    int minorNumber = versionParts[1].toInt();
+
     int minimumMajorNumber = (MINIMUM_GL_VERSION >> 16);
     int minimumMinorNumber = (MINIMUM_GL_VERSION & 0xFF);
-    valid = (majorNumber > minimumMajorNumber
-        || (majorNumber == minimumMajorNumber && minorNumber >= minimumMinorNumber));
+    int majorNumber = 0;
+    int minorNumber = 0;
+    const QString version { "version" };
+    if (version.contains(version)) {
```
Comment (ZappoMan) [inline libraries/gl/src/gl/OpenGLVersionChecker.cpp]: how will this ever be false?

Comment (davidkelly) [inline libraries/gl/src/gl/OpenGLVersionChecker.cpp]: hmm yea -- totally read right by that.  My bad.  We should be checking for whether or not the `'version'` key exists in `glData` I believe.

Comment (samcake) [inline libraries/gl/src/gl/OpenGLVersionChecker.cpp]: That s a good one....
my bad...

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0f39934ec3355a007c5202aa305d9fc324636053 available at: https://gist.github.com/cfaaec64d26b4985982aee1bc347e78c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-0f39934.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-0f39934.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a4e75b4dcf334bcbf9980d187a2cfabbaa730970 available at: https://gist.github.com/cfaaec64d26b4985982aee1bc347e78c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-a4e75b4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-a4e75b4.dmg

**Comment by CaptainOculus** [comment]
testing now on older laptop

**Comment by CaptainOculus** [comment]
image

**Comment by CaptainOculus** [comment]
My apologies, this specific chip supports OpenGL 4.3 and I am in fact shown "Welcome" domain.

**Comment by davidkelly** [comment]
So I believe that there is another bug here -- where we are incorrectly detecting the version.  This fix makes us not crash while detecting, my change currently in another branch (https://github.com/highfidelity/hifi/pull/9037) should fix that.  In fact it makes more sense maybe to combine them into one for testing purposes...

**Comment by samcake** [comment]
Adding the code from the fix in pr 9037 by  @davidkelly

**Comment by hifi-gustavo** [comment]
Build console output for commit: b7381c30cb605ed7e23aa3f7c8d0d773905c5f25 available at: https://gist.github.com/cfaaec64d26b4985982aee1bc347e78c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-b7381c3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-b7381c3.dmg

**Comment by samcake** [comment]
DO NOT MERGE just right now
we are doing a test to be able to repro on intel 530 integrated HD and check the escape sequence

**Comment by hifi-gustavo** [comment]
Build console output for commit: 941e5a57b18c532b683ccd66d729727688d63290 available at: https://gist.github.com/cfaaec64d26b4985982aee1bc347e78c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-941e5a5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-941e5a5.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c01b39b54bdd0fe83fdfb8690a640ae98f1d2cef available at: https://gist.github.com/cfaaec64d26b4985982aee1bc347e78c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-c01b39b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-c01b39b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0ab4a781f59513f598de3e4eaafb38a5a1e1c412 available at: https://gist.github.com/cfaaec64d26b4985982aee1bc347e78c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-0ab4a78.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9027-0ab4a78.dmg

**Pull request merged**